src/App.tsx

```diff
@@ -1,61 +1,160 @@
+};
+
+const validateTitle = (title: string): string => {
+  if (!title || !title.trim()) {
+    return 'Please enter a title';
+  } else {
+    if (!/^[a-zA-Z0-9А-ЩЬЮЯҐЄІЇа-щьюяґєії ]*$/g.test(title)) {
+      return 'Only EN, UA letters, numbers and spaces allowed';
+    }
+  }
+
+  return '';
+};
+
+const validateUser = (userId: number): string => {
+  return userId === 0 ? 'Please choose a user' : '';
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [saveTodo, setSaveTodo] = useState(defaultSaveTodo);
+
+  const [saveTodoError, setSaveTodoError] = useState(defaultSaveTodoError);
+
+  const visibleTodos = getVisibleTodos(todos);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!validateTitle(saveTodo.title) && !validateUser(saveTodo.userId)) {
+      setTodos([
+        ...todos,
+        { id: getNextId(todos), completed: false, ...saveTodo },
+      ]);
+      setSaveTodo(defaultSaveTodo);
+    } else {
+      setSaveTodoError({
+        title: validateTitle(saveTodo.title),
+        user: validateUser(saveTodo.userId),
+      });
+    }
+  };
+
+  const setSaveTodoErrorFn = (
+    field: keyof SaveTodoError,
+    value: string,
+  ): void => {
+    setSaveTodoError(prev => {
+      const newErr = { ...prev };
+
+      newErr[field] = value;
+
+      return newErr;
+    });
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">Title:</label>
+          <input
+            id="title"
+            type="text"
+            data-cy="titleInput"
+            value={saveTodo.title}
+            placeholder="Kangaroo"
+            onChange={event => {
+              if (saveTodoError.title) {
+                setSaveTodoErrorFn('title', '');
+              }
+
+              return setSaveTodo(prev => {
+                return { ...prev, title: event.target.value };
+              });
+            }}
+          />
+          {saveTodoError.title && (
+            <span className="error">{saveTodoError.title}</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <label htmlFor="userId">User:</label>
+          <select
+            id="userId"
+            data-cy="userSelect"
+            value={saveTodo.userId}
+            onChange={event => {
+              {
+                if (saveTodoError.user) {
+                  setSaveTodoErrorFn('user', '');
+                }
+
+                setSaveTodo(prev => {
+                  return { ...prev, userId: +event.target.value };
+                });
+              }
+            }}
```

Comment:
it is not good practice to make anonymous arrow functions in jsx.
better to make function in component and pass reference on it here.